engine/inputdata/district.ftl

```diff
@@ -17,16 +17,55 @@
 
 [#-- Macros to assemble the district configuration --]
 [#macro addDistrict type layers properties=[]  ]
+
     [#local configuration =
         {
             "Type" : type,
             "Properties" : asArray(properties),
-            "Layers" : asArray(layers)
+            "Layers" :
+                getCompositeObject(
+                    [
+                        {
+                            "Names" : "InstanceOrder",
+                            "Description" : "Layers whose configuration data should be included in the solution for an instance of the district",
+                            "Type" : ARRAY_OF_STRING_TYPE,
+                            "Mandatory" : true
+                        },
+                        {
+                            "Names" : "NamePartOrder",
+                            "Description" : "Parts to be included in the name of an instance of the district. If not provided, the InstanceOrder is assumed. If not a defined name part, the part is assumed to be a layer.",
+                            "Type" : ARRAY_OF_STRING_TYPE
+                        },
+                        {
+                            "Names" : "NameParts",
+                            "SubObjects" : true,
+                            "Children" : [
+                                {
+                                    "Names" : "Enabled",
+                                    "Description" : "Should this part be ignored/omitted?",
+                                    "Type" : BOOLEAN_TYPE,
+                                    "Default" : true
+                                },
+                                {
+                                    "Names" : "Fixed",
+                                    "Description" : "Include a fixed string",
+                                    "Type" : STRING_TYPE
+                                },
+                                {
+                                    "Names" : "Ignore",
+                                    "Description" : "An array of values to not include if encountered",
+                                    "Type" : ARRAY_OF_STRING_TYPE
+                                }
+                            ]
+                        }
+                    ],
```

Comment:
Is it possible to make this an Attribute set and then use the attribute set lookup process to get the configuration? That way it can be documented easily with our current schemas?

Reply:
Sure - I didn't go that path because I didn't see it being reused but happy to do it to get doco benefits.

Reply:
Can make it a configuration set to if you want but reference data felt the quickest approach to the configuration available for doco.